device/accton/x86_64-accton_as5835_54x-r0/plugins/sfputil.py

```diff
@@ -288,10 +290,75 @@ def reset(self, port_num):
         
         return True
 
-    def get_transceiver_change_event(self):
-        """
-        TODO: This function need to be implemented
-        when decide to support monitoring SFP(Xcvrd)
-        on this platform.
-        """
-        raise NotImplementedError
+    @property
+    def _get_presence_bitmap(self):
+        nodes = []
+        nodes.append("/sys/bus/i2c/devices/3-0061/module_present_all")
+        nodes.append("/sys/bus/i2c/devices/3-0062/module_present_all")
+
+        bitmap = ""
+        for node in nodes: 
+            try:
+                reg_file = open(node)
+    
+            except IOError as e:
+                print "Error: unable to open file: %s" % str(e)
+                return False
+            bitmap += reg_file.readline().rstrip() + " "
+            reg_file.close()
+
+        rev = bitmap.split(" ")
+        rev.pop() # Remove the last useless character
+        
+        # Convert bitmap into continuously port order
+        rev[4] = hex((int(rev[4],16) | ((int(rev[5],16) & 0x3) << 6)))[2:]      # Port 33-40
+        rev[5] = hex((int(rev[5],16) >> 2) | ((int(rev[6],16) & 0x3) << 6))[2:] # Port 41-48
+
+        # Expand port 49-54
+        tmp = rev.pop()
+        for i in range (2, 8):
+            val = (int(tmp,16) >> i) & 0x1
+            rev.append(hex(val)[2:])
+            
+        for i in range (0,6):
+            rev[i] = rev[i].zfill(2)
+
+        rev = "".join(rev[::-1])
+        return int(rev,16)
+
+    data = {'valid':0, 'last':0, 'present':0} 
+    def get_transceiver_change_event(self, timeout=2000):
```

Comment:
this function is a blocking call and the default timeout is 0 meaning it is blocking.

https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/sonic_sfp/sfputilbase.py#L1257